GpsDataCapturer/sharedlibrary/src/main/java/com/google/sharedlibrary/locationhelper/LocationManagerListener.java

```diff
@@ -24,7 +34,53 @@ public LocationManagerListener(GpsDataCaptureService service) {
     @Override
     public void onGpsStatusChanged(int event) {
         Log.d(TAG, "onGpsStatusChanged");
-        gpsDataCaptureservice.onGpsStatusChanged(event);
+        int satellitesUsedInFix = 0;
+        if (event == GpsStatus.GPS_EVENT_SATELLITE_STATUS) {
+            //Get Gps Status
+            @SuppressLint("MissingPermission")
+            GpsStatus status = gpsDataCaptureservice.getLocationManager().getGpsStatus(null);
+
+            Iterable<GpsSatellite> satellites = status.getSatellites();
+            PriorityQueue<Float> signalPriorityQueue = new PriorityQueue<Float>((a, b) -> Float.compare(b, a));
+
+            int satellitesVisible = 0;
+            float averageSignalStrength = 0;
+            float averageTop4SignalStrength = 0;
+
+            //Get the satellites visible, satellites used in fix and the signal to noise ratio for the satellite.
+            for(GpsSatellite sat: satellites){
+                if(sat.usedInFix()){
+                    satellitesUsedInFix++;
+                    averageSignalStrength += sat.getSnr();
+                    averageTop4SignalStrength = sat.getSnr();
+
+                    if(signalPriorityQueue.size() > 4){
+                        signalPriorityQueue.poll();
+                    }
+                    signalPriorityQueue.add(averageTop4SignalStrength);
+                }
+                satellitesVisible++;
+            }
+
+            //Calculate the average of signal strength and top 4 strongest signal strength
+            if(satellitesUsedInFix != 0){
+                averageSignalStrength /= satellitesUsedInFix;
+                //only take the top 4 strongest signal
+                int size = signalPriorityQueue.size();
+                while(!signalPriorityQueue.isEmpty()){
+                    averageTop4SignalStrength += signalPriorityQueue.poll();
+                }
+                averageTop4SignalStrength /= size;
+            }
+
+            Log.d(TAG, "Satellites visible: " + satellitesVisible);
+            Log.d(TAG, "Satellites used in fix: " + satellitesUsedInFix);
+            Log.d(TAG, "Average of satellites used in fix signal strength: " + averageSignalStrength);
+            Log.d(TAG, "Average of top 4 strongest signal strength: " + averageTop4SignalStrength);
+
+            gpsDataCaptureservice.setAverageOfTop4Signal(averageTop4SignalStrength);
+        }
```

Comment:
could we move this processing to onGpsStatusChanged method in gpsDataCaptureservice

Reply:
Yes, we can move it to the service.